src/UhdmAst.cpp

```diff
@@ -1103,6 +911,206 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
     return nullptr;
 }
 
+AstNode* process_assignment(vpiHandle obj_h, UhdmShared& shared) {
+    AstNode* lvaluep = nullptr;
+    AstNode* rvaluep = nullptr;
+    const unsigned int objectType = vpi_get(vpiType, obj_h);
+    const unsigned int operationType = vpi_get(vpiOpType, obj_h);
+
+    // Right
+    visit_one_to_one({vpiRhs}, obj_h, shared, [&](AstNode* childp) { rvaluep = childp; });
+
+    // Left
+    visit_one_to_one({vpiLhs}, obj_h, shared, [&](AstNode* childp) { lvaluep = childp; });
+
+    const unsigned int arithmeticAssignments[] = {vpiBitAndOp, vpiBitOrOp, vpiBitXorOp,
+                                                  vpiBitXnorOp, vpiLogAndOp, vpiLogOrOp,
+                                                  vpiSubOp, vpiAddOp, vpiMultOp, vpiDivOp,
+                                                  vpiModOp, vpiArithLShiftOp, vpiLShiftOp,
+                                                  vpiRShiftOp, vpiArithRShiftOp};
+    if (std::find(std::begin(arithmeticAssignments),
+                  std::end(arithmeticAssignments),
+                  operationType) != std::end(arithmeticAssignments)) {
+        rvaluep = process_operation(obj_h, shared, {lvaluep, rvaluep});
+        lvaluep = lvaluep->cloneTree(true);
+    }
+
+    if (rvaluep != nullptr && rvaluep->type() == AstType::en::atFOpen) {
+        // Not really an assignpment, AstFOpen node takes care of the lhs
```

Comment:
Typo: assignPment

Reply:
It is copied from previous version of that function. I will correct it.